src/dispatch.rs

```diff
@@ -169,10 +171,12 @@ impl<'t> Dispatcher<'t> {
 
         #[cfg(target_os = "emscripten")]
         self.dispatch_seq(res);
+
+        self.dispatch_thread_local(res);
```

Comment:
I'd just like to avoid having any more contracts here. Could we instead have two methods, `wait()` and `wait_without_tl()` (or a better name) so that valid API usage already makes it correct in this case?